packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2230,3 +2608,334 @@ function escapeJSStringsForInstructionScripts(input: string): string {
+  destination: Destination,
+  resources: Resources,
+  responseState: ResponseState,
+): boolean {
+  const explicitPreloadsTarget = [];
+  const remainingTarget = [];
+
+  const {precedences, explicitPreloads, implicitPreloads} = resources;
+
+  // Flush stylesheets first by earliest precedence
+  precedences.forEach(precedenceResources => {
+    precedenceResources.forEach(resource => {
+      // resources should not already be flushed so we elide this check
+      pushLinkImpl(remainingTarget, resource.props, responseState);
+      resource.flushed = true;
+      resource.inShell = true;
+      resource.hint.flushed = true;
+    });
+  });
+
+  explicitPreloads.forEach(resource => {
+    if (!resource.flushed) {
+      pushLinkImpl(explicitPreloadsTarget, resource.props, responseState);
+      resource.flushed = true;
+    }
+  });
+  explicitPreloads.clear();
+
+  implicitPreloads.forEach(resource => {
+    if (!resource.flushed) {
+      pushLinkImpl(remainingTarget, resource.props, responseState);
+      resource.flushed = true;
+    }
+  });
+  implicitPreloads.clear();
+
+  let i;
+  let r = true;
+  for (i = 0; i < explicitPreloadsTarget.length - 1; i++) {
+    writeChunk(destination, explicitPreloadsTarget[i]);
+  }
+  if (i < explicitPreloadsTarget.length) {
+    r = writeChunkAndReturn(destination, explicitPreloadsTarget[i]);
+  }
+
+  for (i = 0; i < remainingTarget.length - 1; i++) {
+    writeChunk(destination, remainingTarget[i]);
+  }
+  if (i < remainingTarget.length) {
+    r = writeChunkAndReturn(destination, remainingTarget[i]);
+  }
+  return r;
+}
+
+export function writeImmediateResources(
+  destination: Destination,
+  resources: Resources,
+  responseState: ResponseState,
+): boolean {
+  const {explicitPreloads, implicitPreloads} = resources;
+  const target = [];
+
+  explicitPreloads.forEach(resource => {
+    if (!resource.flushed) {
+      pushLinkImpl(target, resource.props, responseState);
+      resource.flushed = true;
+    }
+  });
+  explicitPreloads.clear();
+
+  implicitPreloads.forEach(resource => {
+    if (!resource.flushed) {
+      pushLinkImpl(target, resource.props, responseState);
+      resource.flushed = true;
+    }
+  });
+  implicitPreloads.clear();
+
+  let i = 0;
+  for (; i < target.length - 1; i++) {
+    writeChunk(destination, target[i]);
+  }
+  if (i < target.length) {
+    return writeChunkAndReturn(destination, target[i]);
+  }
+  return false;
+}
+
+function hasStyleResourceDependencies(
+  boundaryResources: BoundaryResources,
+): boolean {
+  const iter = boundaryResources.values();
+  // At the moment boundaries only accumulate style resources
+  // so we assume the type is correct and don't check it
+  while (true) {
+    const {value: resource} = iter.next();
+    if (!resource) break;
+
+    // If every style Resource flushed in the shell we do not need to send
+    // any dependencies
+    if (!resource.inShell) {
+      return true;
+    }
+  }
```

Comment:
In contrast to https://github.com/facebook/react/pull/25243#discussion_r981856789

Should I keep the iter here since I am essentially trying to recreate Array.some for the map and want to bail out before total iteration?